include/xtensor-sparse/xcoo_container.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef XSPARSE_COO_CONTAINER_HPP
+#define XSPARSE_COO_CONTAINER_HPP
+
+#include "xsparse_container.hpp"
+
+namespace xt
+{
+    /******************************
+     * xcoo_container declaration *
+     ******************************/
+
+    template <class D>
+    class xcoo_container : public xsparse_container<D>
+    {
+    public:
+        using base_type = xsparse_container<D>;
+        using index_storage_type = typename base_type::index_storage_type;
+        using storage_type = typename base_type::storage_type;
+        using index_type = typename base_type::index_type;
+        using value_type = typename base_type::value_type;
+        using reference = typename base_type::reference;
+        using const_reference = typename base_type::const_reference;
+        using pointer = typename base_type::pointer;
+        using const_pointer = typename base_type::const_pointer;
+        using shape_type = typename base_type::shape_type;
+        using inner_shape_type = typename base_type::inner_shape_type;
+        using size_type = typename base_type::size_type;
+
+        using true_pointer = value_type*;
+
+        const index_storage_type& index_storage() const noexcept;
+        index_storage_type& index_storage() noexcept;
+
+        true_pointer find_element(const index_type& index);
+        void insert_element(const index_type& index, const_reference value);
+        void remove_element(const index_type& index);
```

Comment:
These methods should be in the private section and `xsparse_reference` should  be added as a friend class.

Reply:
Why not protected as `update_entries` and `access_impl`?

Reply:
Except if these methods are required in the inheriting classes, they all should be private.